src/runtime/security/client_negotiation.cpp

```diff
@@ -142,6 +142,33 @@ void client_negotiation::on_mechanism_selected(const negotiation_response &resp)
     }
 }
 
+void client_negotiation::on_challenge(const negotiation_response &challenge)
+{
+    if (challenge.status == negotiation_status::type::SASL_CHALLENGE) {
+        std::string response_msg;
+        auto err = _sasl->step(challenge.msg, response_msg);
+        if (!err.is_ok() && err.code() != ERR_SASL_INCOMPLETE) {
```

Comment:
should `||`？

Reply:
No, both `err.is_ok()` and `err.code() == ERR_SASL_INCOMPLETE` means the `_sasl->step()` is ok